src/Models/ModelInstantiator.php

```diff
@@ -36,21 +36,27 @@ public function arrayToDataObject(
 
         $instance = new $class();
 
+        $property_reflection = $this->dataObjectProperties($class);
+
         // This holds information about OneToMany and ManyToOne relationships.
-        $cardinalities = $this->dataObjectPropertyCardinalities($class);
+        $cardinalities = $this->dataObjectPropertyCardinalities($property_reflection);
 
         // Assign public properties.
-        foreach ($this->dataObjectProperties($class) as $property)
+        foreach ($property_reflection as $reflection)
         {
+            $property = $reflection->getName();
+
             // Ignore properties which are not specified in the incoming array,
             // and properties which define relationships to other Data Objects.
             if (
                 $enforcer->propIsDefined($arr, $property) &&
                 !array_key_exists($property, $cardinalities)
             )
             {
-                $property_reflection = new ReflectionProperty($class, $property);
-                $instance->{$property} = $property_reflection->getType()->getName() === 'DateTime' ? new DateTime($arr[$property]) : $arr[$property];
+                $instance->{$property} = $enforcer->propIsNotNull($arr, $property)
```

Comment:
Two expressions in a ternary is a bit too compressed. For these core components of our API, we should strive for readability as much as possible.